subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishDependenciesIntegTest.groovy

```diff
@@ -158,11 +158,15 @@ class MavenPublishDependenciesIntegTest extends AbstractIntegrationSpec {
 
         where:
         plugin         | gradleConfiguration  | mavenScope
-        'java'         | 'compile'            | 'runtime'
+        'java'         | 'compile'            | 'compile'
+        'java'         | 'runtime'            | 'compile'
```

Comment:
So it seems like we're trading one confusing behavior for another.  So this code snippet:
```groovy
dependencies {
    compile 'org.apache.commons:commons-math3:3.6.1'
    runtime 'com.google.guava:guava:21.0'
}
```
will yield three different results.
With the java plugin, before 4.0: both dependencies are 'runtime' scoped
With the java-library plugin (3.4 and 3.5), the compile dependency is 'compile' scoped and the runtime dependency is 'runtime' scoped.
With either the java plugin or java-library plugin with 4.0, both are 'runtime' scoped.

It seems like the current behavior with the java-library plugin is the best.  Although you indicate it is inconsistent with multi-project builds.  What's the rationale for making 'runtime' dependencies exposed in the POM as 'compile' scoped?